tooling/greybox_fuzzer/src/corpus.rs

```diff
@@ -0,0 +1,431 @@
+    pub fn id(&self) -> TestCaseId {
+        self.id
+    }
+
+    /// Creates a new TestCase with a specified ID and input values
+    pub fn with_id(id: TestCaseId, value: &'a InputMap) -> Self {
+        Self { id, value }
+    }
+}
+
+impl<'a> From<&'a InputMap> for TestCase<'a> {
+    /// Converts an InputMap into a TestCase by generating a new unique ID
+    fn from(value: &'a InputMap) -> Self {
+        Self { value, id: generate_testcase_id() }
+    }
+}
+
+/// Manages reading and writing corpus files to the filesystem, including parsing JSON inputs
+pub struct CorpusFileManager {
+    // Manages file operations and tracks file metadata
+    file_manager: FileManager,
+    // Path to the directory containing corpus files
+    corpus_path: PathBuf,
+    // ABI definition used for parsing inputs
+    abi: Abi,
+    // Maps file IDs to their parsed input values
+    parsed_map: HashMap<FileId, InputMap>,
+}
+
+impl CorpusFileManager {
+    /// Creates a new CorpusFileManager for a specific package and test harness
+    pub fn new(root: &Path, package_name: &str, harness_name: &str, abi: Abi) -> Self {
+        let corpus_path = root.join(package_name).join(harness_name);
+
+        Self { file_manager: FileManager::new(root), corpus_path, abi, parsed_map: HashMap::new() }
+    }
+
+    /// Loads all corpus files from disk into memory, parsing them as JSON inputs.
+    /// Creates the corpus directory if it doesn't exist.
+    /// Returns an error if directory creation fails or if any files can't be parsed.
+    pub fn load_corpus_from_disk(&mut self) -> Result<(), String> {
+        let mut builder = DirBuilder::new();
+        match builder.recursive(true).create(&self.corpus_path) {
+            Ok(_) => (),
+            Err(_) => {
+                return Err(format!(
+                    "Couldn't create directory {:?}",
+                    self.corpus_path.as_os_str()
+                ));
+            }
+        }
+        // Go through all files in the corpus directory recursively
+        for entry in WalkDir::new(&self.corpus_path) {
+            let Ok(entry) = entry else {
+                continue;
+            };
+            if !entry.file_type().is_file() {
+                continue;
+            }
+            if !entry.path().extension().map_or(false, |ext| ext == CORPUS_FILE_EXTENSION) {
+                continue;
+            };
+            let path = entry.into_path();
+            // Skip files that are already loaded
+            if self.file_manager.has_file(&path) {
+                continue;
+            }
+            let source = std::fs::read_to_string(path.as_path())
+                .unwrap_or_else(|_| panic!("could not read file {:?} into string", path));
+
+            let parsed_source = parse_json(&source, &self.abi).map_err(|parsing_error| {
+                format!("Error while parsing file {:?}: {:?}", path.as_os_str(), parsing_error)
+            })?;
```

Comment:
If you run `nargo fuzz` against one function and then change that function's signature and run `nargo fuzz` again it's almost sure you'll get an error here:

```
[one] Executed fuzzing task on test...issue with corpus:
Error while parsing file "corpus/one/test/060a619ba2124046f9b324b52ba4526b7896ed40ed7b2bc52a962eac1a178637.json": MissingArgument("y")
[one] 1 test failed
```

I guess that's fine, but I wonder if we can guide the user towards fixing that error. For example I deleted that file and it failed again because there was another file. After deleting both files it succeeded.

Not something to improve here, but maybe capture a follow-up issue for this.

Reply:
Yes, I will fix in a follow-up PR